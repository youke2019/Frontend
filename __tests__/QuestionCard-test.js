import React from 'react'
import renderer from 'react-test-renderer'
import {shallow, mount, render, configure} from 'enzyme'
import QuestionCard from '../src/components/QuestionCard'
import Adapter from "enzyme-adapter-react-16/build"


configure({adapter: new Adapter()})

const question = {
        "question_id": 17,
        "user_id": "01231",
        "course_id": "66974",
        "question_content": "老师好看吗",
        "question_time": "2019-07-18 20:44:34",
        "question_isbanned": false,
        "question_praise_point": 2,
        "courseAnswerList": [
            {
                "answer_id": 33,
                "question_id": 17,
                "user_id": "1A7A1F80-59D0-4E8C-B4C6-AAAFFC606120",
                "answer_content": "杜鹃老师很好看",
                "answer_time": "2019-07-18 22:20:42",
                "answer_isbanned": false,
                "answer_praise_point": 0,
                "current_user_praise": false,
                "courseAnswerPraiseList": []
            },
            {
                "answer_id": 37,
                "question_id": 17,
                "user_id": "01231",
                "answer_content": "没用的问题",
                "answer_time": "2019-07-19 11:06:35",
                "answer_isbanned": false,
                "answer_praise_point": 0,
                "current_user_praise": false,
                "courseAnswerPraiseList": []
            },
            {
                "answer_id": 38,
                "question_id": 17,
                "user_id": "01231",
                "answer_content": "楼上说啥呢",
                "answer_time": "2019-07-19 13:22:44",
                "answer_isbanned": false,
                "answer_praise_point": 0,
                "current_user_praise": false,
                "courseAnswerPraiseList": []
            }
        ],
        "courseQuestionPraiseList": [
            {
                "question_praise_id": 1757,
                "question_id": 17,
                "user_id": "1A7A1F80-59D0-4E8C-B4C6-AAAFFC606120"
            },
            {
                "question_praise_id": 1758,
                "question_id": 17,
                "user_id": "01231"
            }
        ],
        "current_user_praise": false
}

const flush = jest.fn(() => {})
jest.mock('axios')
global.baseUrl = 'baseUrl'

describe('functions', () => {
    let wrapper = shallow(
        <QuestionCard
            onAnswer={flush}
            QandA = {question}
        />)

    wrapper.setProps({
        QandA:question
    })

    it('displayInput()', () => {
        expect(wrapper.instance().state.answer_visible).toBe(false)
        wrapper.instance().displayInput()
        expect(wrapper.instance().state.answer_visible).toBe(true)
    })

    it('hideInput()', () => {
        expect(wrapper.instance().state.answer_visible).toBe(true)
        wrapper.instance().hideInput()
        expect(wrapper.instance().state.answer_visible).toBe(false)
    })

    it('useful()', (done) => {
        expect(wrapper.instance().props.QandA.current_user_praise).toBe(false)
        expect(wrapper.instance().props.QandA.question_praise_point).toBe(2)
        wrapper.instance().useful()

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.instance().props.QandA.current_user_praise).toBe(true)
                expect(wrapper.instance().props.QandA.question_praise_point).toBe(3)

                wrapper.instance().useful()

                const promise = new Promise(
                    function (resolve, reject)
                    {resolve('success')})
                promise
                    .then(data=>{
                        expect(wrapper.instance().props.QandA.current_user_praise).toBe(false)
                        expect(wrapper.instance().props.QandA.question_praise_point).toBe(2)
                        done()
                    })
                    .catch(err=>{console.log(err)})
            })
            .catch(err=>{console.log(err)})
    })

    it('praise()', (done) => {
        expect(wrapper.instance().props.QandA.courseAnswerList[0].current_user_praise).toBe(false)
        expect(wrapper.instance().props.QandA.courseAnswerList[0].answer_praise_point).toBe(0)
        wrapper.instance().praise(0)

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.instance().props.QandA.courseAnswerList[0].current_user_praise).toBe(true)
                expect(wrapper.instance().props.QandA.courseAnswerList[0].answer_praise_point).toBe(1)

                wrapper.instance().praise(0)

                const promise = new Promise(
                    function (resolve, reject)
                    {resolve('success')})
                promise
                    .then(data=>{
                        expect(wrapper.instance().props.QandA.courseAnswerList[0].current_user_praise).toBe(false)
                        expect(wrapper.instance().props.QandA.courseAnswerList[0].answer_praise_point).toBe(0)
                        done()
                    })
                    .catch(err=>{console.log(err)})
            })
            .catch(err=>{console.log(err)})
    })

    it('praise() annoymous', (done) => {
        expect(wrapper.instance().props.QandA.courseAnswerList[0].current_user_praise).toBe(false)
        expect(wrapper.instance().props.QandA.courseAnswerList[0].answer_praise_point).toBe(0)
        wrapper.find('TouchableOpacity').at(1).props().onPress()

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.instance().props.QandA.courseAnswerList[0].current_user_praise).toBe(true)
                expect(wrapper.instance().props.QandA.courseAnswerList[0].answer_praise_point).toBe(1)
                done()
            })
            .catch(err=>{console.log(err)})
    })

    it('answer() normal', (done) => {
        wrapper.setState({
            answer_visible: true
        })
        expect(wrapper.instance().state.answer_visible).toBe(true)
        wrapper.instance().answer()

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.instance().state.answer_visible).toBe(false)
                expect(flush).toHaveBeenCalled()
                done()
            })
            .catch(err=>{console.log(err)})
    })

    it('answer() abnormal', (done) => {
        wrapper.setState({
            answer_visible: true
        })
        expect(wrapper.instance().state.answer_visible).toBe(true)
        wrapper.instance().answer('error')

        const promise = new Promise(
            function (resolve, reject)
            {resolve('success')})
        promise
            .then(data=>{
                expect(wrapper.instance().state.answer_visible).toBe(true)
                done()
            })
            .catch(err=>{console.log(err)})
    })

    console.log(wrapper.instance())
})

test('renders QuestionCard page correctly', () => {
    const tree = renderer.create(
        <QuestionCard
        onAnswer={flush}
        QandA = {question}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});