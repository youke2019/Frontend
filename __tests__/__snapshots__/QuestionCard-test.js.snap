// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders QuestionCard page correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 30,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "borderColor": "lightgrey",
          "borderTopWidth": 0.2,
          "bottom": 0,
          "height": 50,
          "paddingHorizontal": 10,
          "paddingVertical": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        keyboardShouldPersistTaps="always"
        style={
          Object {
            "contentSize": 0,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "whitesmoke",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "orange",
                  "borderRadius": 8,
                  "borderWidth": 0.5,
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                autoFocus={true}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="点击输入回复"
                rejectResponderTermination={true}
                style={
                  Object {
                    "fontSize": 15,
                    "padding": 2,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1.5,
                  "marginVertical": 0,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FDD326",
                    "borderRadius": 8,
                    "height": "auto",
                    "marginHorizontal": 8,
                    "paddingVertical": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "字魂70号-灵悦黑体",
                      "fontSize": 15,
                      "lineHeight": 15,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                   发送
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 20,
        "elevation": 4,
        "paddingLeft": 30,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingRight": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          问题
        </Text>
        <Image
          source={
            Object {
              "uri": "question_mark",
            }
          }
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "flex": 5,
            "fontSize": 14,
            "fontWeight": "500",
            "paddingHorizontal": 5,
          }
        }
      >
        老师好看吗
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingRight": 15,
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={
            Object {
              "uri": "bulb",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "width": 20,
              },
              Object {
                "tintColor": "#FDAF26",
              },
            ]
          }
        />
        <Text
          style={
            Object {
              "fontSize": 8,
              "paddingHorizontal": 4,
            }
          }
        >
          有用(
          3
          )
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 6,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingLeft": 8,
            "paddingVertical": 10,
            "width": 260,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            回答
          </Text>
          <Image
            source={
              Object {
                "uri": "answer_mark",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "flex": 6,
              "fontSize": 13,
              "paddingHorizontal": 10,
            }
          }
        >
          杜鹃老师很好看
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="stretch"
            source={
              Object {
                "uri": "dianzan",
              }
            }
            style={
              Array [
                Object {
                  "height": 12,
                  "width": 12,
                },
                Object {
                  "tintColor": "#D3D3D3",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontSize": 8,
                "paddingHorizontal": 4,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingLeft": 8,
            "paddingVertical": 10,
            "width": 260,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            回答
          </Text>
          <Image
            source={
              Object {
                "uri": "answer_mark",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "flex": 6,
              "fontSize": 13,
              "paddingHorizontal": 10,
            }
          }
        >
          没用的问题
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="stretch"
            source={
              Object {
                "uri": "dianzan",
              }
            }
            style={
              Array [
                Object {
                  "height": 12,
                  "width": 12,
                },
                Object {
                  "tintColor": "#D3D3D3",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontSize": 8,
                "paddingHorizontal": 4,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingLeft": 8,
            "paddingVertical": 10,
            "width": 260,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            回答
          </Text>
          <Image
            source={
              Object {
                "uri": "answer_mark",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "flex": 6,
              "fontSize": 13,
              "paddingHorizontal": 10,
            }
          }
        >
          楼上说啥呢
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="stretch"
            source={
              Object {
                "uri": "dianzan",
              }
            }
            style={
              Array [
                Object {
                  "height": 12,
                  "width": 12,
                },
                Object {
                  "tintColor": "#D3D3D3",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontSize": 8,
                "paddingHorizontal": 4,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "height": 60,
          "justifyContent": "center",
          "width": 240,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "width": 110,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "button_orange",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 40,
                  "width": 110,
                },
                Object {
                  "resizeMode": "stretch",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontFamily": "字魂95号-手刻宋",
                "fontSize": 14,
                "letterSpacing": 1,
                "paddingHorizontal": 5,
                "paddingTop": 2,
              }
            }
          >
            我来回答
          </Text>
          <Image
            source={
              Object {
                "uri": "to_answer",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders home page correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 30,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "borderColor": "lightgrey",
          "borderTopWidth": 0.2,
          "bottom": 0,
          "height": 50,
          "paddingHorizontal": 10,
          "paddingVertical": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        keyboardShouldPersistTaps="always"
        style={
          Object {
            "contentSize": 0,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "whitesmoke",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "orange",
                  "borderRadius": 8,
                  "borderWidth": 0.5,
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                autoFocus={true}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="点击输入回复"
                rejectResponderTermination={true}
                style={
                  Object {
                    "fontSize": 15,
                    "padding": 2,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1.5,
                  "marginVertical": 0,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FDD326",
                    "borderRadius": 8,
                    "height": "auto",
                    "marginHorizontal": 8,
                    "paddingVertical": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "字魂70号-灵悦黑体",
                      "fontSize": 15,
                      "lineHeight": 15,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                   发送
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 20,
        "elevation": 4,
        "paddingLeft": 30,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingRight": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          问题
        </Text>
        <Image
          source={
            Object {
              "uri": "question_mark",
            }
          }
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "flex": 5,
            "fontSize": 14,
            "fontWeight": "500",
            "paddingHorizontal": 5,
          }
        }
      >
        老师好看吗
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingRight": 15,
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={
            Object {
              "uri": "bulb",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "width": 20,
              },
              Object {
                "tintColor": "#FDAF26",
              },
            ]
          }
        />
        <Text
          style={
            Object {
              "fontSize": 8,
              "paddingHorizontal": 4,
            }
          }
        >
          有用(
          2
          )
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 6,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingLeft": 8,
            "paddingVertical": 10,
            "width": 260,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            回答
          </Text>
          <Image
            source={
              Object {
                "uri": "answer_mark",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "flex": 6,
              "fontSize": 13,
              "paddingHorizontal": 10,
            }
          }
        >
          杜鹃老师很好看
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="stretch"
            source={
              Object {
                "uri": "dianzan",
              }
            }
            style={
              Array [
                Object {
                  "height": 12,
                  "width": 12,
                },
                Object {
                  "tintColor": "#D3D3D3",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontSize": 8,
                "paddingHorizontal": 4,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingLeft": 8,
            "paddingVertical": 10,
            "width": 260,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            回答
          </Text>
          <Image
            source={
              Object {
                "uri": "answer_mark",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "flex": 6,
              "fontSize": 13,
              "paddingHorizontal": 10,
            }
          }
        >
          没用的问题
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="stretch"
            source={
              Object {
                "uri": "dianzan",
              }
            }
            style={
              Array [
                Object {
                  "height": 12,
                  "width": 12,
                },
                Object {
                  "tintColor": "#D3D3D3",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontSize": 8,
                "paddingHorizontal": 4,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingLeft": 8,
            "paddingVertical": 10,
            "width": 260,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            回答
          </Text>
          <Image
            source={
              Object {
                "uri": "answer_mark",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "flex": 6,
              "fontSize": 13,
              "paddingHorizontal": 10,
            }
          }
        >
          楼上说啥呢
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="stretch"
            source={
              Object {
                "uri": "dianzan",
              }
            }
            style={
              Array [
                Object {
                  "height": 12,
                  "width": 12,
                },
                Object {
                  "tintColor": "#D3D3D3",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontSize": 8,
                "paddingHorizontal": 4,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "height": 60,
          "justifyContent": "center",
          "width": 240,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "width": 110,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "button_orange",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 40,
                  "width": 110,
                },
                Object {
                  "resizeMode": "stretch",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "fontFamily": "字魂95号-手刻宋",
                "fontSize": 14,
                "letterSpacing": 1,
                "paddingHorizontal": 5,
                "paddingTop": 2,
              }
            }
          >
            我来回答
          </Text>
          <Image
            source={
              Object {
                "uri": "to_answer",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
